applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_2D2N.cpp

```diff
@@ -971,6 +971,36 @@ int CrBeamElement2D2N::Check(const ProcessInfo& rCurrentProcessInfo) const
     KRATOS_CATCH("")
 }
 
+const Parameters CrBeamElement2D2N::GetSpecifications() const
+{
+    const Parameters specifications = Parameters(R"({
+        "time_integration"           : ["static","implicit","explicit"],
+        "framework"                  : "lagrangian",
+        "symmetric_lhs"              : true,
+        "positive_definite_lhs"      : true,
+        "output"                     : {
+            "gauss_point"            : ["MOMENT","FORCE","INTEGRATION_COORDINATES"],
+            "nodal_historical"       : ["DISPLACEMENT","ROTATION","VELOCITY","ACCELERATION"],
+            "nodal_non_historical"   : [],
+            "entity"                 : []
+        },
+        "required_variables"         : ["DISPLACEMENT","ROTATION"],
+        "required_dofs"              : ["DISPLACEMENT_X","DISPLACEMENT_Y","ROTATION_Z"],
+        "flags_used"                 : [],
+        "compatible_geometries"      : ["Line2D2"],
+        "element_integrates_in_time" : false,
+        "compatible_constitutive_laws": {
+            "type"        : ["BeamConstitutiveLaw"],
+            "dimension"   : ["2D"],
+            "strain_size" : [1]

```

Comment:
@clazaro and @KlausBSautter for a 2D beam the strain size should be 2 (sigma_xx, tau_xy) and in 3D should be 3 (sigma_xx, tau_xy, tau_xz) ?

In any case we should also discuss in general the meaning of the dimension for a CL (are shells 3D¿?, are trusses 1D¿?)

Reply:
Disclaimer: still a Kratos newbie... That said, the Beam Constitutive Law is an empty law: the constitutive behavior is defined within the element code and terms of generalized stresses. The results at integration points are: (N, V2, M3) in the case of the 2D beam, and (N, V2, V3, Mt, M2, M3) in the case of the 3D beam. There is no evaluation of stresses at material points of the cross section.

Reply:
So shall we keep 1 as it is right now?

Reply:
I'd vote for:

2D beam: CL dimension = 2 ; strain_size = 3
3D beam: CL dimension = 3 ; strain_size = 6

Although the meaning of 'dimension' is just inherited from the embedding space.

Reply:
> Although the meaning of 'dimension' is just inherited from the embedding space.

Thanks @clazaro, I was thinking this

> 2D beam: CL dimension = 2 ; strain_size = 3
> 3D beam: CL dimension = 3 ; strain_size = 6

For the strain size I was thinking in the neutral axis, but material points make more sense

Reply:
Thanks  @clazaro. I'll change it accordingly.